crates/ide/src/references/rename.rs

```diff
@@ -241,62 +250,78 @@ fn rename_mod(
 
     let mut source_change = SourceChange::default();
 
-    let src = module.definition_source(sema.db);
-    let file_id = src.file_id.original_file(sema.db);
-    match src.value {
-        ModuleSource::SourceFile(..) => {
-            // mod is defined in path/to/dir/mod.rs
-            let path = if module.is_mod_rs(sema.db) {
-                format!("../{}/mod.rs", new_name)
-            } else {
-                format!("{}.rs", new_name)
-            };
-            let dst = AnchoredPathBuf { anchor: file_id, path };
-            let move_file = FileSystemEdit::MoveFile { src: file_id, dst };
-            source_change.push_file_system_edit(move_file);
-        }
-        ModuleSource::Module(..) => {}
-    }
-
-    if let Some(src) = module.declaration_source(sema.db) {
-        let file_id = src.file_id.original_file(sema.db);
-        let name = src.value.name().unwrap();
-        source_change.insert_source_edit(
-            file_id,
-            TextEdit::replace(name.syntax().text_range(), new_name.into()),
-        );
+    let InFile { file_id, value: def_source } = module.definition_source(sema.db);
+    let file_id = file_id.original_file(sema.db);
+    if let ModuleSource::SourceFile(..) = def_source {
+        // mod is defined in path/to/dir/mod.rs
+        let path = if module.is_mod_rs(sema.db) {
+            format!("../{}/mod.rs", new_name)
+        } else {
+            format!("{}.rs", new_name)
+        };
+        let dst = AnchoredPathBuf { anchor: file_id, path };
+        let move_file = FileSystemEdit::MoveFile { src: file_id, dst };
+        source_change.push_file_system_edit(move_file);
+    }
+
+    if let Some(InFile { file_id, value: decl_source }) = module.declaration_source(sema.db) {
+        let file_id = file_id.original_file(sema.db);
+        match decl_source.name() {
+            Some(name) => source_change.insert_source_edit(
+                file_id,
+                TextEdit::replace(name.syntax().text_range(), new_name.to_string()),
+            ),
+            _ => unreachable!(),
+        };
     }
-
-    let RangeInfo { range, info: refs } = find_all_refs(sema, position)?;
-    let ref_edits = refs.references().iter().map(|(&file_id, references)| {
+    let def = Definition::ModuleDef(ModuleDef::Module(module));
+    let usages = def.usages(sema).all();
+    let ref_edits = usages.iter().map(|(&file_id, references)| {
         source_edit_from_references(sema, file_id, references, new_name)
     });
     source_change.extend(ref_edits);
 
-    Ok(RangeInfo::new(range, source_change))
+    Ok(RangeInfo::new(TextRange::default(), source_change))
```

Comment:
This PR changes all the `TextRange`s to default, reason being that they aren't actually needed anymore. They were only needed for the `prepareRename` request which doesn't use these functions anymore. I plan to do a follow up PR to this that just removes the RangeInfo usage here instead to keep the diff smaller.